tools/expotools/src/commands/AndroidBuildPackages.ts

```diff
@@ -0,0 +1,345 @@
+    readline.cursorTo(process.stdout, 0);
+    process.stdout.write(` ✅  Finished building ${pkg.name}\n`);
+  }
+
+  await _restoreFilesAsync(filesToStash);
+
+  console.log(' 🚚  Copying newly built packages...');
+
+  await spawnAsync('mkdir', ['-p', path.join(ANDROID_DIR, 'maven/com/facebook')])
+  await spawnAsync('mkdir', ['-p', path.join(ANDROID_DIR, 'maven/host/exp/exponent')]);
+  await spawnAsync('mkdir', ['-p', path.join(ANDROID_DIR, 'maven/org/unimodules')]);
+
+  for (const pkg of packages) {
+    await spawnAsync('cp', [
+      '-r',
+      path.join(process.env.HOME!, '.m2', 'repository', pkg.buildDirRelative),
+      path.join(ANDROID_DIR, 'maven', pkg.buildDirRelative),
+    ]);
+  }
+
+  // Copy JSC
+  await spawnAsync('rm', ['-rf', path.join(ANDROID_DIR, 'maven/org/webkit/')]);
+  await spawnAsync('cp', [
+    '-r',
+    path.join(ANDROID_DIR, '../node_modules/jsc-android/dist/org/webkit'),
+    path.join(ANDROID_DIR, 'maven/org/webkit/'),
+  ]);
+}
+
+async function action(options: ActionOptions) {
+  if (!options.sdkVersion) {
+    throw new Error('Must run with `--sdkVersion SDK_VERSION`');
+  }
+
+  const detachableUniversalModules = await _findUnimodules(path.join(EXPO_ROOT_DIR, 'packages'));
+
+  // packages must stay in this order --
+  // expoview MUST be last
+  const packages: Package[] = [
+    REACT_ANDROID_PKG,
+    ...detachableUniversalModules,
+    EXPOVIEW_PKG,
+  ];
+  let packagesToBuild: string[] = [];
+
+  const expoviewBuildGradle = await fs.readFile(
+    path.join(ANDROID_DIR, 'expoview', 'build.gradle'),
+  );
+  const match = expoviewBuildGradle.toString().match(/api 'com.facebook.react:react-native:([\d.]+)'/);
+  if (!match[1]) {
+    throw new Error('Could not find SDK version in android/expoview/build.gradle: unexpected format');
+  }
+
+  if (match[1] !== options.sdkVersion) {
+    console.log(" 🔍  It looks like you're adding a new SDK version. Ignoring the `--packages` option and rebuilding all packages...");
+    packagesToBuild = packages.map(pkg => pkg.name);
+  } else if (options.packages) {
+    if (options.packages === 'all') {
+      packagesToBuild = packages.map(pkg => pkg.name);
+    } else if (options.packages === 'suggested') {
+      console.log(' 🔍  Gathering data about packages...');
+      packagesToBuild = await _getSuggestedPackagesToBuild(packages);
+    } else {
+      const packageNames = options.packages.split(',');
+      packagesToBuild = packages.map(pkg => pkg.name).filter(pkgName => packageNames.includes(pkgName));
+    }
+    console.log(' 🛠   Rebuilding the following packages:');
+    console.log(packagesToBuild);
+  } else {
+    // gather suggested package data and then show prompts
+    console.log(' 🔍  Gathering data...');
+
+    packagesToBuild = await _getSuggestedPackagesToBuild(packages);
+
+    console.log(' 🕵️   It appears that the following packages need to be rebuilt:');
+    console.log(packagesToBuild);
+
+    const { option } = await inquirer.prompt([{
```

Comment:
Does it compile well with TypeScript? Imho you should do this:
```suggestion
    const { option } = await inquirer.prompt<{ option: string }>([{
```